src/tarballs/build.ts

```diff
@@ -72,186 +72,194 @@ const copyCoreYarnFiles = async (yarnRootPath: string, workspacePath: string) =>
 }
 
-export async function build(
-  c: BuildConfig,
-  options: {
-    pack?: boolean
-    parallel?: boolean
-    platform?: string
-    tarball?: string
-  } = {},
-): Promise<void> {
-  const {config, xz} = c
-  const packCLI = async () => {
-    const {stdout} = await exec('npm pack --unsafe-perm', {cwd: c.root})
-    return path.join(c.root, stdout.trim().split('\n').pop()!)
-  }
+type BuildOptions = {
+  pack?: boolean
+  parallel?: boolean
+  platform?: string
+  preserveLockfiles?: boolean
+  tarball?: string
+}
 
-  const extractCLI = async (tarball: string) => {
-    await emptyDir(c.workspace())
-    const tarballNewLocation = path.join(c.workspace(), path.basename(tarball))
-    await move(tarball, tarballNewLocation)
-    let tarCommand = `tar -xzf "${tarballNewLocation}"`
-    if (process.platform === 'win32') tarCommand += ' --force-local'
-    await exec(tarCommand, {cwd: c.workspace()})
-    const files = await readdir(path.join(c.workspace(), 'package'), {withFileTypes: true})
-    await Promise.all(
-      files.map((i) => move(path.join(c.workspace(), 'package', i.name), path.join(c.workspace(), i.name))),
-    )
-
-    await Promise.all([
-      rm(path.join(c.workspace(), 'package'), {recursive: true}),
-      rm(path.join(c.workspace(), path.basename(tarball)), {recursive: true}),
-      remove(path.join(c.workspace(), 'bin', 'run.cmd')),
-    ])
+export async function build(c: BuildConfig, options: BuildOptions = {}): Promise<void> {
+  log(`gathering workspace for ${c.config.bin} to ${c.workspace()}`)
+  await extractCLI(options.tarball ?? (await packCLI(c)), c)
+  await updatePJSON(c)
+  await addDependencies(c)
+  await writeBinScripts({
+    baseWorkspace: c.workspace(),
+    config: c.config,
+    nodeOptions: c.nodeOptions,
+    nodeVersion: c.nodeVersion,
+  })
+  await pretarball(c)
+  if (!options.preserveLockfiles) {
+    await removeLockfiles(c)
   }
 
-  const updatePJSON = async () => {
-    const pjsonPath = path.join(c.workspace(), 'package.json')
-    const pjson = await readJSON(pjsonPath)
-    pjson.version = config.version
-    pjson.oclif.update = pjson.oclif.update || {}
-    pjson.oclif.update.s3 = pjson.oclif.update.s3 || {}
-    pjson.oclif.update.s3.bucket = c.s3Config.bucket
-    await writeJSON(pjsonPath, pjson, {spaces: 2})
+  const targetsToBuild = c.targets.filter((t) => !options.platform || options.platform === t.platform)
+  if (options.parallel) {
+    log(`will build ${targetsToBuild.length} targets in parallel`)
+    await Promise.all(targetsToBuild.map((t) => buildTarget(t, c, options)))
+  } else {
+    log(`will build ${targetsToBuild.length} targets sequentially`)
+    for (const target of targetsToBuild) {
+      // eslint-disable-next-line no-await-in-loop
+      await buildTarget(target, c, options)
+    }
+
+    log(`finished building ${targetsToBuild.length} targets sequentially`)
   }
+}
+
+const isLockFile = (f: string) =>
```

Comment:
`pnpm-lock.yaml`?

Reply:
cool, didn't know about that one, will add.